modules/src/ics03_connection/version.rs

```diff
@@ -70,22 +70,26 @@ pub fn get_compatible_versions() -> Vec<Version> {
 pub fn pick_version(
     supported_versions: Vec<Version>,
     counterparty_versions: Vec<Version>,
-) -> Option<Version> {
+) -> Result<Version, Error> {
     let mut intersection: Vec<Version> = vec![];
     for s in supported_versions.iter() {
         for c in counterparty_versions.iter() {
             if c.identifier != s.identifier {
                 continue;
             }
-            // TODO - perform feature intersection and error if empty
+            for feature in c.features.iter() {
```

Comment:
I think the idea here was to return an error if there are no common features between `c` and `s`. With a `HashSet`, that would be as simple as `c.features.intersection(&s.features).collect().is_empty()`.